### PR TITLE
Merge didcomm into master

### PR DESCRIPTION
**Pull request by joshmann35**
Implementation of DIDCommV2 in the pico engine.

### CODE REVIEW

File: pico-engine
Comment (joshmann35) [inline pico-engine:1]: I don't know what this is and why it is here...

File: packages/pico-engine/src/index.ts
Comment (joshmann35) [inline packages/pico-engine/src/index.ts:1]: This adds the did-o ruleset to automatically be installed to root pico

File: packages/pico-engine/krl/io.picolabs.wrangler.krl [hunk 1]
```diff
@@ -1,473 +1,484 @@
...
+        parts = ctx:rid_url.split("/")
+        parts.splice(parts.length()-1,1,did_o_rid+".krl").join("/")
+      }
+      did_o_url = did_o_ruleset_url()
+    }
+    fired {
+      raise wrangler event "install_ruleset_request"
+        attributes {"url":subs_url}
+      raise wrangler event "install_ruleset_request"
+        attributes {"url":did_o_url}
```
Comment (joshmann35) [inline packages/pico-engine/krl/io.picolabs.wrangler.krl:422]: These changes install the did-o ruleset automatically to child picos.

File: packages/pico-engine-core/src/PicoEngineCore.ts
Comment (joshmann35) [inline packages/pico-engine-core/src/PicoEngineCore.ts:1]: These changes adds the dido module to be accessible to rulesets.

File: packages/pico-engine-core/package.json
Comment (joshmann35) [inline packages/pico-engine-core/package.json:1]: Adds the didcomm-node library from sicpa-dlab to manage packing/unpacking didcommv2 messages

**Comment by b1conrad** [comment]
Josh, do you have any thoughts on why the first automated check failed?

**Comment by joshmann35** [comment]
> Josh, do you have any thoughts on why the first automated check failed?

I don't know. The tests in the main branch fail as well...

**Pull request merged**